src/main/Duke/virtue/Command.java

```diff
@@ -0,0 +1,135 @@
+package virtue;
+
+public class Command {
+    // The available types of commands in Virtue.
+    public enum CommandType {
+        BYE ("bye", false, false),
+        LIST ("list", false, false),
+        MARK ("mark", true, false),
+        UNMARK ("unmark", true, false),
+        TODO ("todo", false, true),
+        DEADLINE ("deadline", false, true),
+        EVENT ("event", false, true),
+        DELETE("delete", true, false);
+
+        private final String string;
+        private final boolean hasIndex;
+        private final boolean hasDescription;
+
+        CommandType(String string,
+                    boolean hasIndex,
+                    boolean hasDescription) {
+            this.string = string;
+            this.hasIndex = hasIndex;
+            this.hasDescription = hasDescription;
+        }
+
+        @Override
+        public String toString() {
+            return this.string;
+        }
+    }
+
+    private String command;
+    protected CommandType type;
+    protected int index;
+    protected String description;
+    protected String by;
+    protected String from;
+    protected String to;
+
+    public Command(String command) throws VirtueException {
+        this.command = command;
+        this.type = getCommandType();
+
+        if (this.type.hasIndex) {
+            this.index = getIndex();
+        }
+
+        if (this.type.hasDescription) {
+            this.description = getDescription();
+        }
+
+        if (this.type == CommandType.DEADLINE) {
+            this.by = getBy();
+        }
+
+        if (this.type == CommandType.EVENT) {
+            this.from = getFrom();
+            this.to = getTo();
+        }
+    }
+
+    // Gets the type of the command, which is its first word.
+    private CommandType getCommandType() throws UnknownCommandTypeException {
+        String firstWord = command.split(" ", 2)[0];
+
+        for (CommandType type : CommandType.values()) {
+            if (firstWord.equals(type.toString())) {
+                return type;
+            }
+        }
+
+        throw new UnknownCommandTypeException();
```

Comment:
using specific exceptions makes error handling more precise and helps with debugging